src/aro/Parser.zig

```diff
@@ -2488,7 +2489,12 @@ fn enumSpec(p: *Parser) Error!Type {
             try p.errTok(.enum_fixed, colon);
             break :fixed null;
         };
-        try p.errTok(.enum_fixed, colon);
+        if (fixed.isInt() and !fixed.is(.@"enum")) {
+            try p.errTok(.enum_fixed, colon);
+        } else {
+            try p.errStr(.invalid_type_underlying_enum, ty_start, try p.typeStr(fixed));
+            break :fixed Type.int;
+        }
```

Comment:
I think this keeps it clear that the `.enum_fixed` warning is meant to be issued always.
```suggestion

        if (!fixed.isInt() or fixed.is(.@"enum")) {
            try p.errStr(.invalid_type_underlying_enum, ty_start, try p.typeStr(fixed));
            break :fixed Type.int;
        }

        try p.errTok(.enum_fixed, colon);
```